File: lib/csharp/se/lth/control/labcomm/LabCommDispatcher.cs
namespace se.lth.control.labcomm {

  using System;

  public interface LabCommDispatcher {
    
    Type getSampleClass();
    
    String getName();
    
    byte[] getSignature();
    
    void decodeAndHandle(LabCommDecoder decoder,
			 LabCommHandler handler);

  }

}

File: lib/java/se/lth/control/labcomm2006/LabCommDispatcher.java
package se.lth.control.labcomm2006;

public interface LabCommDispatcher {
    
  public Class getSampleClass();
    
  public String getName();

  public byte[] getSignature();

  public void decodeAndHandle(LabCommDecoder decoder,
			      LabCommHandler handler) throws Exception;

}

